client/src/components/QnA/QuestionsList.jsx

```diff
@@ -20,22 +25,71 @@ class QuestionsList extends React.Component {
     });
   }
 
+  // show modal method
+  showQuestionModal() {
+    this.props.sendInteraction('Question Modal');
+    this.setState({
+      showModal: true
+    });
+  }
+
+  // handle cancel of modal
+  handleCancel() {
+    this.setState({
+      showModal: false
+    });
+  }
+
   componentDidMount() {
 
     // Destructing props
     const {formatBody} = this.props;
     const params =
      {
-       product_id: 47425,
+       product_id: 47422,
        page: 1,
        count: 5
      };
     const body = formatBody('GET', '/qa/questions', params);
 
     axios.post('/api/*', body)
       .then((results) => {
-        this.setState({
-          questionList: [...results.data.results]
+        // this.setState({
+        //   questionList: [...results.data.results]
+        // });
+        this.setState((state) => {
+          return {
+            questionList: [...results.data.results]
+          };
+        }, () => {
+          // call api to get all the questions of that particular product
+          let questionListResult = [];
+          const getAllQuestions = async () => {
+            let pageCount = 1;
+
+            while (true) {
```

Comment:
Interesting function, Ash.   Does  " while(true) "  mean while there are still pages being returned?

Reply:
yes. As the api doesn't tell across how many pages it has the results, I need to run this logic until any page returns empty list of questions.